packages/client/hmi-client/src/components/dataset/Dataset.vue

```diff
@@ -108,39 +108,56 @@
 				<template #header>
 					Data preview<span class="artifact-amount">({{ csvContent?.length }} rows)</span>
 				</template>
+
 				<DataTable
 					tableStyle="width:auto"
 					class="p-datatable-sm"
-					:value="csvContent"
+					:value="csvContent?.slice(1, csvContent.length)"
 					removableSort
 					resizable-columns
 					showGridlines
 				>
 					<Column
-						v-for="colName of rawColumnNames"
-						:key="colName"
-						:field="colName"
+						v-for="(colName, index) of csvHeaders"
+						:key="index"
+						:field="index.toString()"
 						:header="colName"
 						sortable
 					>
+						<!-- column summary charts below -->
+						<template #header>
+							<div class="histogram">
+								<div class="histogram-label-min">Min: {{ csvMinsToDisplay?.at(index) }}</div>
+								<Chart
+									type="bar"
+									:height="800"
+									:data="chartData?.at(index)"
+									:options="chartOptions"
+								/>
+								<div class="histogram-label-max">Max: {{ csvMaxsToDisplay?.at(index) }}</div>
+								<div class="histogram-label-other">Mean: {{ csvMeansToDisplay?.at(index) }}</div>
+								<div class="histogram-label-other">Median: {{ csvMedianToDisplay?.at(index) }}</div>
+								<div class="histogram-label-other">SD: {{ csvSdToDisplay?.at(index) }}</div>
```

Comment:
All of this and it's attached JS and CSS should be on its own component.
Even if its only used by this dataset, it will make Dataset more compact.

Reply:
Good point